storage/store_mgr.go

```
@@ -15,79 +15,246 @@
+		name := strings.ToLower(storeTypeName)
+		if !collections.ContainsString(checkStores, name) {
+			log.Printf("Default config store will be used for store type:%q.", storeTypeName)
+			var defaultStore config.Store
+			switch storeTypeName {
+			case types.StoreTypeDeployment.String():
+				defaultStore = config.Store{
+					Name:           "defaultFileStoreWithCache",
+					Implementation: fileStoreWithCacheImpl,
+					Types:          []string{types.StoreTypeDeployment.String()},
+				}
+			case types.StoreTypeEvent.String():
+				defaultStore = config.Store{
+					Name:           "defaultConsulStore" + types.StoreTypeEvent.String(),
+					Implementation: consulStoreImpl,
+					Types:          []string{types.StoreTypeEvent.String()},
+				}
+			case types.StoreTypeLog.String():
+				defaultStore = config.Store{
+					Name:           "defaultConsulStore" + types.StoreTypeLog.String(),
+					Implementation: consulStoreImpl,
+					Types:          []string{types.StoreTypeLog.String()},
+				}
+			}
+
+			cfgStores = append(cfgStores, defaultStore)
 		}
 	}
+	return cfgStores
+}
 
+// Create store implementations
+func createStoreImpl(cfg config.Configuration, configStore config.Store, storeType types.StoreType) error {
+	var err error
+	impl := strings.ToLower(configStore.Implementation)
+	switch impl {
+	case strings.ToLower(fileStoreWithCacheImpl), strings.ToLower(fileStoreWithCacheAndEncryptionImpl):
+		stores[storeType], err = file.NewStore(cfg, configStore.Name, configStore.Properties, true, impl == strings.ToLower(fileStoreWithCacheAndEncryptionImpl))
+		if err != nil {
+			return err
+		}
+	case strings.ToLower(consulStoreImpl):
+		stores[storeType] = consul.NewStore()
+	default:
+		log.Printf("[WARNING] unknown store implementation:%q. This will be ignored.", storeType)
+	}
 	return nil
 }
 
 // GetStore returns the store related to a defined store type
-func GetStore(typ types.StoreType) Store {
+func GetStore(typ types.StoreType) store.Store {
```

Maybe use check existence form (the one that also returns a bool) of the get key in map to validate WDYT?

---

Makes sense

---

But at this point, as each store is mandatory, we should panic if the store is missing

---

Or just return the bool if later we handle optional stores ?

---

But it makes the code more complex for the caller:
``

---

https://github.com/golang/go/wiki/CodeReviewComments#dont-panic

---

This is not a "normal" error as the stores have to exist. Panic is relevant in this case.

In the future, if we handle optional store, we may have to change this but not sure to need it ?

---

:+1: